internal/node/node.go

```diff
@@ -355,7 +359,7 @@ func (n *Node) DoElection() bool {
 		if maxTermSeen > n.Term {
 			log.Info().
 				Int64("max response term", maxTermSeen).
-				Str("other node", maxTermSeenSource).
+				Str("other node", maxTermSeenSource.Id).
```

Comment:
This causes a nil pointer dereference when a node comes up and starts an election but then gets a 'nay' vote from a node with a later term. It seems like it shouldn't, but it does. *shrug*

I commented out this line and everything behaves as desired, including the redirects with both `curl` and the browser

Reply:
We fixed this by propagating the node as part of a NodeReply in cd9025e. We also added a test for this so we don't get caught out in future